src/impl/tpls/KokkosSparse_spmv_tpl_spec_decl.hpp

```diff
@@ -70,6 +71,84 @@ struct SPMV<class AT, class AO, class AD, class AM, class AS,
     const YVector& y) {
   Kokkos::Profiling::pushRegion("KokkosSparse::spmv[TPL_CUSPARSE,double]");
   printf("Using CUSPARSE for spmv!\n");
+
+  cudaError_t      cuError;
+  cusparseStatus_t status;
+  cusparseHandle_t handle=0;
+  // cusparseMatDescr_t descr=0;
+
+  /* initialize cusparse library */
+  status = cusparseCreate(&handle);
+  if (status != CUSPARSE_STATUS_SUCCESS) {
+    throw("KokkosSparse::spmv[TPL_CUSPARSE,double]: cusparse was not initialized correctly");
```

Comment:
1. Please don't throw `const char[]`.  Use a subclass of `std::exception`.
2. Kokkos should have a function for reporting errors; please use that instead of throwing.

Reply:
Once things are actually working (right now the spmv does not give the right answer), I'll look into the proper way of throwing exception or reporting errors. We must have some mechanism for that in other tpl interfaces, maybe in cublas_gemm interface for instance?

Reply:
@mhoemmen 
I looked into `Kokkos_Cuda_Error.hpp` and the main blocker to use `CUDA_SAFE_CALL` is that it uses the type `cudaError` when the cusparse calls are retunring `cusparseStatus_t` instead. At this point I added better throws that use `std::runtime_exception`.
I will add to the todo list the creation of a `CUSPARSE_SAFE_CALL` macro that replicates mechanism demonstrated in Kokkos but for cusparse return type.